### DISCUSSION TITLE
Using and customize Elsa studio components in angular 2+

### DISCUSSION THREAD

**Discussion by omar-louati** [elsa-workflows/elsa-core#3000] [Q&A]
Hello,

Is it possible to use the Elsa studio components (version 2.) in angular 2+ and also customize them to add new functionalities like a resume button when an instance is suspended or change background colors depending on the activity status...

Thanks in advance.

**Comment by jdevillard** [comment]
Hello, 

it is possible to use the Elsa Studio Component as a WebComponent if you use directly the npm package. 
You can then customize the CSS if you want to change some element like as you mention a background color.

it's also possible to extend some capability of the designer with adding of some control around the activity like icon, design-time behavior or field type like it's mentions on the following documentation : https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-designer-plugins

If you want to add other specific functionality, you can fork and work directly on the source code. In this case, you have then two possibilities : 
- use the dist and load the studio as a WebComponent (like how it is available in the NPM package)
- add some angular integration to package easily the studio like it's explain in the stencils documentation : https://stenciljs.com/docs/angular

HTH
-

**Reply by jdevillard** [reply]
you can also take a look at this repo https://github.com/jdevillard/ElsaWorkflowDashboard-Angular
